app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -0,0 +1,46 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.theme.GreenFavoriteButtton
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+fun onIconClick() {
+    TODO("Not yet implemented")
+}
+
+@Composable
+fun FavoriteButton(
+    isFavorite: Boolean,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+) {
+    Image(
+        painter = painterResource(id = if (isFavorite) R.drawable.ic_favorite_filled else R.drawable.ic_favorite),
+        contentDescription = null,
+        modifier = modifier
+            .clickable { onClick() }
+            .size(32.dp)
+    )
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun FavoriteButtonPreview() {
+    val favoriteButtonModifier = Modifier.padding(5.dp)
+    FavoriteButton(
+        isFavorite = false,
+        modifier = favoriteButtonModifier,
+        onClick = { onIconClick() }
```

Comment:
Here you should create `isFavorite` state and then pass its value to `isFavorite` parameter and mutate it in onClick.
Something like this:
```
val isFavoriteState  = remember { mutableStateOf(false) } 
FavoriteButton(
   isFavorite = /*pass value here */,
   modifier = favoriteButtonModifier,
   onClick = { /* mutate value of state here */ }
) 
```